Sergei Perepelkin - Week 2

SergPerep opened the pull request.

acadavid reviewed (changes requested):
@SergPerep Good job with the homework so far! Can you please correct the first part of the exercise where you created a new table to relate mentors? The rest looks good! Read the comments for some additional thoughts.

Once you correct the rist part of the homework, I will approve it.

SergPerep commented on Week2/mysql-exercises/db.js:
```diff
@@ -0,0 +1,14 @@
+const db = mysql.createConnection({
+  host: "localhost",
+  user: "hyfuser",
+  password: "HyF2022/FluSH1916",
```
I'll move it to `.env`

acadavid replied:
Oh no, it's fine here for the exerise, you're just the one who is changing the password so I appreciate it :)

SergPerep commented on Week2/mysql-exercises/queries/createRelationTables.js:
```diff
@@ -0,0 +1,36 @@
+
+  // PUBLISHMENT
+  db.query(
+    `CREATE TABLE publishment (
```
It does sound a little weird, and I'm not completely happy with it, but it seems like naming relational tables is generally a challenge.

acadavid replied:
100%

SergPerep commented:
- Moved hostname, username, and password to `.env`
- Removed mentorship table and put mentor id into authors table
- Replaced all `WITH` statements with chaining `SELECT`

acadavid commented:
Great job! You can also simplify your queries by removing the selects in the middle, for example:

```sql
SELECT authors.id as 'Id', name as 'Author', university_id, date_of_birth, h_index, gender, mentor, title 
AS 'Paper' FROM authors 
LEFT JOIN publishment
LEFT JOIN papers ON publishment.paper_id = papers.id
ON authors.id = publishment.author_id;
```
